crates/duty-tracker/src/contract_manager.rs

```diff
@@ -1027,48 +942,90 @@ impl ContractManagerCtx {
 
-                        duties.extend(
-                            contract
-                                .process_event(ContractEvent::RootSig(key.clone(), *sig))
-                                .await?,
-                        );
-                    } else {
-                        warn!(txid=%txid, "received signatures exchange for unknown session - contract may not be indexed yet");
-                    }
+                    duties.extend(
+                        contract.process_contract_event(ContractEvent::RootSig(key, *sig))?,
+                    );
                 }
 
                 Ok(duties)
             }
         }
     }
 
+    /// Processes incoming P2P requests from other operators and generates appropriate duty
+    /// responses.
+    ///
+    /// This method handles different types of P2P requests and generates the corresponding operator
+    /// duties that need to be executed in response. The method supports the following request
+    /// types:
+    ///
+    /// # Request Types
+    ///
+    /// - [`GetMessageRequest::StakeChainExchange`]: Requests for stake chain exchange data
+    ///   - Returns an [`OperatorDuty::PublishStakeChainExchange`] duty
+    ///
+    /// - [`GetMessageRequest::DepositSetup`]: Requests for deposit setup information for a specific
+    ///   contract
+    ///   - Validates that the requested contract exists in active contracts
+    ///   - Returns an [`OperatorDuty::PublishDepositSetup`] duty with deposit and stake chain
+    ///     information
+    ///
+    /// - [`GetMessageRequest::Musig2NoncesExchange`]: Requests for MuSig2 nonces for either graph
+    ///   or root transactions
+    ///   - For claim transactions: Returns an [`OperatorDuty::PublishGraphNonces`] duty with pegout
+    ///     graph data
+    ///   - For deposit request transactions: Returns an [`OperatorDuty::PublishRootNonce`] duty
+    ///     with deposit transaction data
+    ///
+    /// - [`GetMessageRequest::Musig2SignaturesExchange`]: Requests for MuSig2 partial signatures
+    ///   - For claim transactions: Returns an [`OperatorDuty::PublishGraphSignatures`] duty with
+    ///     graph signature data
+    ///   - For deposit request transactions: Returns an [`OperatorDuty::PublishRootSignature`] duty
+    ///     with root signature data
+    ///
+    /// # Arguments
+    ///
+    /// * `req` - The P2P request from another operator
+    ///
+    /// # Returns
+    ///
+    /// Returns a vector of [`OperatorDuty`] instances that should be executed in response to the
+    /// request. An empty vector indicates no duties are needed (e.g., when contract is not
+    /// found or not ready).
+    ///
+    /// # Errors
+    ///
+    /// Returns [`ContractManagerErr::InvalidP2PRequest`] if:
+    ///
+    /// - A deposit setup request is made for a non-existent contract
+    /// - The request format is invalid or malformed
     async fn process_p2p_request(
```

Comment:
This function is too long

Reply:
Ok take a look at the refactor in 51f049746c0780b4c963d033edab5d4e12712525